crates/valence_entity/src/active_status_effects.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+
+use bevy_ecs::prelude::*;
+
+use crate::status_effects::StatusEffect;
+
+/// [`Component`] that stores the [`ActiveStatusEffect`]s of an [`Entity`].
+#[derive(Component, Default, Debug)]
+pub struct ActiveStatusEffects {
+    active_effects: HashMap<StatusEffect, ActiveStatusEffect>,
```

Comment:
I don't see why this is a hash-map as you don't randomly access based on `StatusEffect`. You only iterate it (something witch a hash-map is slow at). I would recommend a `Vec<(StatusEffect, ActiveStatusEffect)>` or maybe a `BTreeMap`.

Reply:
I'm pretty certain I want some sort of a map just because I **don't** want any duplicates.

Reply:
A vec would suit it as we can only have like 30 different effects and checking is a vec contains less then 50 is basicly costless. Infact I would remove the useless data and just store some kind of numerical id to check againsed . a Vec is the fastest data structure to iterate.

Reply:
Will the client display duplicate effects? eg could you have 2 speed effects

Reply:
no that's impossible. the client will just replace the old effect

Reply:
It's more important that we don't have inconsistencies with client state. Stick with a map.

Reply:
@dyc3 Could you look at the new code? I can't see how it can have inconsistencies with the client state.